src/libraries/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs

```diff
@@ -233,5 +234,22 @@ public static void SubclassAssemblyLoadContext_Properties()
             Assert.Contains(alc, AssemblyLoadContext.All);
             Assert.Empty(alc.Assemblies);
         }
+
+        class RefEmitLoadContext : AssemblyLoadContext
+        {
+            protected override Assembly? Load(AssemblyName assemblyName)
+            {
+                return AssemblyBuilder.DefineDynamicAssembly(assemblyName, AssemblyBuilderAccess.Run);
+            }
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsReflectionEmitSupported))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/31804", TestRuntimes.Mono)]
+        public static void LoadRefEmitAssembly()
```

Comment:
I have added Assert to verify that `Resolving` event handle is not called. 

We treat the misbehaving callbacks as hard errors that terminate the assembly loading. I think it would look better if these hard errors are not wrapped in FileNotFoundException, but it is not something that I plan to change in this PR.